ethstore/src/dir/mod.rs

```diff
@@ -62,6 +63,8 @@ pub trait KeyDirectory: Send + Sync {
 	fn path(&self) -> Option<&PathBuf> { None }
 	/// Return vault provider, if available
 	fn as_vault_provider(&self) -> Option<&VaultKeyDirectoryProvider> { None }
+	/// Returns hash of the directory content, if supported
+	fn hash(&self) -> Result<H256, Error> { Ok(H256::zero()) }
```

Comment:
Even so, there should not be a valid implementation of a key directory which causes `reload_if_changed` to think the data has never changed.

If we broaden the description of the function to something like "Returns a unique 32-byte representation of the directory's contents.", then I don't imagine any implementation which couldn't do it.

(in that case, a more descriptive name like `unique_repr(&self) -> Result<H256, Error>` might make sense)